### ISSUE TITLE
Allow for meta definition in the HTML head

### ISSUE THREAD

**Issue by arnaudbreton** [mjmlio/mjml#82]
Following up on issue #19, we want to provide MJML users with a way to define meta information in the HTML `head` section. The idea here is to discuss openly here and to settle on a final implementation by the end of the week.

In the new `mj-head` tag introduced (see #78), the behaviour of `mj-raw` tags will be to output its content in the final rendered HTML. No validation is done here by the MJML rendering engine. 

``` html
<mj-head>
    <mj-raw>
        ...
    </mj-raw>
</mj-head>
```

**Comment by debiasej** [comment]
I would like to change the meta tag. For this I add a `mj-raw` tag in the `mj-head`but it doesn't seem to work.
```
<mj-head
    <mj-raw>
        <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" />
    </mj-raw>
</mj-head>
```

Is it possible to do that?

**Comment by Marvin256** [comment]
Same thing for me, I would like to add this for Outlook 2013-2016 in the <head> to avoid Times New Roman font when using custom font-family:

**Comment by iRyusa** [comment]
It's in the mid term roadmap for MJML, you check it here : https://github.com/mjmlio/mjml/projects/1#card-635717

@Marvin256 It's an other issue but it shouldn't fallback to Times New Roman when using a custom font. Did you try to set default font-family attribute with `mj-attributes` & `mj-all`?

**Comment by Marvin256** [comment]
Thank you for your answer @iRyusa, mj-all isn't working for this particular issue. See this post from Litmus for more information : https://litmus.com/community/discussions/982-outlook-overrides-font-to-times-new-roman

Just tried to add a mj-raw in the mj-container and Outlook is aplying the good font.

Here's the code for those having the issue : 

```
      <mj-raw>
        <!--[if mso]>
        <style type="text/css" media="all">
          body, table, td, span {font-family: Helvetica, Arial, sans-serif !important;}
        </style>
        <![endif]-->
      </mj-raw>
```

**Comment by iRyusa** [comment]
@Marvin256 It should, I just tried with Ubuntu : https://mjml.io/try-it-live/BJP0kzzWg and it renders as Arial on Outlook. 

It only fallbacks to Times when custom font are imported. In MJML we're excluding every custom fonts imported for outlook with conditionnal comment to avoid this behaviour, so `font-family` will be applied correctly

**Comment by dimitarnestorov** [comment]
@debiasej I was able to do it, though quite painfully and poorly
Not sure how you run MJML, but I'm using Node and I can simply register a new MJ Head element
```
var mjml = require('mjml');
mjml.registerMJHeadElement("mj-meta", function(element, args){
	args.container = args.container.replace("</head>", '<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=0">');
});
```
And then inside the head I added
```
<mj-head>
	...

	<mj-meta />
</mj-head>
```

**Comment by iRyusa** [comment]
You should add a </head> at the end of your replace @dimitarnestorov but that's a way to fix it atm

**Comment by dimitarnestorov** [comment]
@iRyusa 
Thanks for the heads up.
I looked at the result though. Has a `</head>` even though I missed it 😅
In that case, you've made on hell of a framework!

**Comment by dalefish** [comment]
Quick hack for this:

v3.3.0 and above

<mj-style>
</style>
<meta ....>
<meta ....>
</mj-style>


Kind Regards,

:----------------------------:
Dale McConnell

**Comment by matiasm** [comment]
We had been using the above `</style>` hack to add things to our head, but this broke in at least `4.1.1` because of the parser - and even `4.0.5` (which we had been using) has its dependencies on `4.1.1+` so we couldn't simply pin our version to `4.0.5`.

A hack to the hack is to perform the same pattern but on the title tag:

```
<mj-title>
  {{ title }}
  </title>
  <!-- just <head> things -->
</mj-title>
```

which doesn't seem to break the parser in the same way as the original hack. Hopefully helps anyone wanting to use this before the `<mj-raw>` option is available!

**Comment by kmcb777** [comment]
Hi @matiasm you can now use mj-raw in the mj-head so this hack isn't needed anymore in the new release mjml4.2-beta, which you can install with npm install mjml

**Comment by donni106** [comment]
Sorry, no change after removing and adding mjml again.

**Comment by iRyusa** [comment]
Just tried with a clean install of mjml and it works just fine. Can you detail a bit your setup ( do you have mjml installed globally ? ), node version, npm/yarn versions, os, the way you compile mjml files ?

**Comment by tkafka** [comment]
I just got the same with 4.3, it used to work with 4.2:

```
mj-raw cannot be used inside mj-head, only inside: mj-column, mj-attributes, mj-body, mj-group, mj-section, mj-wrapper
```

In my `package.json` I have
```
  "dependencies": {
    "mjml": "~4.3.0"
  },
```

And here's a relevant section of `yarn.lock` - all installed components of mjml are 4.3.0:
```

mjml-accordion@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-accordion/-/mjml-accordion-4.3.0.tgz#a7dce26f8de2788a8e409c6004c89391d60ccfa0"
  integrity sha512-XccD03MdWzH/GcIxk6u0KEAtlj6g8CDuYJz11JS2QzHtyCXO7TG4YGpBP+T8NEevTSxRBIqiWdn8QzTQ0NNIKQ==
  dependencies:
    babel-runtime "^6.26.0"
    cross-env "^5.1.4"
    lodash "^4.17.2"
    mjml-core "4.3.0"

mjml-body@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-body/-/mjml-body-4.3.0.tgz#939d85101efd1894cec7c65efb7095a48579ac7f"
  integrity sha512-7080fYt6+c82ZAtTHofOU6qDEZ8a1dsqf0xsd5AYpLQ4ndLtDp0Or+92rLXRV+FzMDD7k4Wdg80H71XVRkPKjw==
  dependencies:
    babel-runtime "^6.26.0"
    cross-env "^5.1.4"
    lodash "^4.17.2"
    mjml-core "4.3.0"

mjml-button@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-button/-/mjml-button-4.3.0.tgz#25797d363cac2cd2131168006d111b219de5b1e3"
  integrity sha512-L2jbGQRm3/6BYuQKYt9bmDD8Cih4M4Q964xzb9M+ViiV0v4zlbSZz4epaebSptv5+NeGJkOivRzoIQxTB4Tlvw==
  dependencies:
    babel-runtime "^6.26.0"
    cross-env "^5.1.4"
    lodash "^4.17.2"
    mjml-core "4.3.0"

mjml-carousel@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-carousel/-/mjml-carousel-4.3.0.tgz#66156a630f2e6ec5af76e3c0cc3b5294113eee33"
  integrity sha512-YwxUDf7qxipunYDvF7Ee5Ldc797gihhnfM15Y6u2Qk1ZxFNbSQHCFhl+py6QLMCvz5k7y1LwFo9RAwI+kGg9Hg==
  dependencies:
    babel-runtime "^6.26.0"
    cross-env "^5.1.4"
    lodash "^4.17.2"
    mjml-core "4.3.0"

mjml-cli@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-cli/-/mjml-cli-4.3.0.tgz#3728e2d03db61d7ba76934600e7df8e8f34c1a8a"
  integrity sha512-Vw9Ev7ECx+wxqfFjAL3lM1msp6/rCvcE7nghWY83uBuJUxCd575py2NXpy576gENakBCixuvLZODgwsDg5ZSZg==
  dependencies:
    babel-runtime "^6.26.0"
    chokidar "^1.6.1"
    cross-env "^5.1.4"
    glob "^7.1.1"
    lodash "^4.17.4"
    mjml-core "4.3.0"
    mjml-migrate "4.3.0"
    mjml-parser-xml "4.3.0"
    mjml-validator "4.3.0"
    yargs "^8.0.2"

mjml-column@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-column/-/mjml-column-4.3.0.tgz#1a80b58a8a2ec416a2085e1a452cc196709cb698"
  integrity sha512-XQhDjNEw5WvPQt1wbDEFMS443SxsmmGnJdxE9fIGw9uD9uMLm+VSAAicKKkuy2cOWQhwDqnjpmVcXHPe1JlTbA==
  dependencies:
    babel-runtime "^6.26.0"
    cross-env "^5.1.4"
    lodash "^4.17.2"
    mjml-core "4.3.0"

mjml-core@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-core/-/mjml-core-4.3.0.tgz#1e2452ac5202a58ddf21e647607cf4c4ecff30e2"
  integrity sha512-8SZWS1MwDF6+7SIVZaWzgewUDLwF4cJiTZ5BLQ1SSdoRXTQ1QS/06rEXUR0mm6D4Q9Y1eQYa6B5vS43SXswTTQ==
  dependencies:
    babel-runtime "^6.26.0"
    cross-env "^5.1.4"
    html-minifier "^3.5.3"
    js-beautify "^1.6.14"
    juice "^4.1.0"
    lodash "^4.17.2"
    mjml-migrate "4.3.0"
    mjml-parser-xml "4.3.0"
    mjml-validator "4.3.0"

mjml-divider@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-divider/-/mjml-divider-4.3.0.tgz#3a4196a7ee66b99ad654bc02f02811e7ff208ea1"
  integrity sha512-hmcU52f+ncb78Zsakx6zn5YbLkxb5zMrbv5+OEqB3K5niBrFfJwEnWGtASoMGvUT1w/rR7oIc3FsvTShEVD6RA==
  dependencies:
    babel-runtime "^6.26.0"
    cross-env "^5.1.4"
    lodash "^4.17.2"
    mjml-core "4.3.0"

mjml-group@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-group/-/mjml-group-4.3.0.tgz#58915a00efca553e8993b86bad7483ec3d39be38"
  integrity sha512-Huve8+wVLMO3WlkaB4MSXB0hCz9/X/KPaPZzyek24t9luup4NZLwTcxg/8vMdDhfHEnH7M1a5a44EdPHyg3CJA==
  dependencies:
    babel-runtime "^6.26.0"
    cross-env "^5.1.4"
    lodash "^4.17.2"
    mjml-core "4.3.0"

mjml-head-attributes@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-head-attributes/-/mjml-head-attributes-4.3.0.tgz#100fe58df2e4845d5966165c8fcb1b8e0ec0f658"
  integrity sha512-rQ+Eo7v48j2ZxwKlCZ2nEQ3ReJzO3sUGJWqFUMSU8Ssl6jZLLoe5+GVGeVLguTCRrALDTXGXJ/SJurM58BU23Q==
  dependencies:
    babel-runtime "^6.26.0"
    cross-env "^5.1.4"
    lodash "^4.17.2"
    mjml-core "4.3.0"

mjml-head-breakpoint@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-head-breakpoint/-/mjml-head-breakpoint-4.3.0.tgz#0cbcfa2df0e37d5cd46c0e0d8e5c67eeceff4707"
  integrity sha512-o4Krb5Hq8wu7K1RXBygcJPP3Fsodl+J8z5UNPDNH0i6rnuECgrtUFUIBoYhT263bbTOqrJD51iHP2VaFH9vdhw==
  dependencies:
    babel-runtime "^6.26.0"
    cross-env "^5.1.4"
    lodash "^4.17.2"
    mjml-core "4.3.0"

mjml-head-font@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-head-font/-/mjml-head-font-4.3.0.tgz#3551875b195fbfd3b65c3aadda9b7e4b31998584"
  integrity sha512-FUILgXFVD6UicdixK811vC98rEi4W3omhumVLEdH4yKsr9H+3wDUcVr1P02ulM2dMDfPbUTXgdDzSTEgKfUK+Q==
  dependencies:
    babel-runtime "^6.26.0"
    cross-env "^5.1.4"
    lodash "^4.17.2"
    mjml-core "4.3.0"

mjml-head-preview@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-head-preview/-/mjml-head-preview-4.3.0.tgz#1cc90d58f2ed3163a7082f15192404c5ddbe2efe"
  integrity sha512-GlxF2/UdiXi3gGa13ZnGWIFoFzBFfU1pVmvx02ug2PN29nSaTNGvbYah4u2e4Lzqte83ZE1Jf834IPHnYgvbwA==
  dependencies:
    babel-runtime "^6.26.0"
    cross-env "^5.1.4"
    lodash "^4.17.2"
    mjml-core "4.3.0"

mjml-head-style@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-head-style/-/mjml-head-style-4.3.0.tgz#8a8d1089d66993a212a76114d0fbf5d209bbd209"
  integrity sha512-rhFiIESWd5a/UPBDpgUxHFVjXuRTvjc6+x2COBFQNRqMOIQ/PXnNfvVbvmIvH2FcAHZLPC+pGd4GtOtKiqtNWw==
  dependencies:
    babel-runtime "^6.26.0"
    cross-env "^5.1.4"
    lodash "^4.17.2"
    mjml-core "4.3.0"

mjml-head-title@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-head-title/-/mjml-head-title-4.3.0.tgz#73fe748387546e1497a8f98baf61017a8bf38832"
  integrity sha512-PaFkq0S9t8zJqjlVLu5RlYbLkUE71oTj+gv2jYhCvgdgOeDRJNGkm3WGYI812B08hIkiUsWSserpbjbvP4kayw==
  dependencies:
    babel-runtime "^6.26.0"
    cross-env "^5.1.4"
    lodash "^4.17.2"
    mjml-core "4.3.0"

mjml-head@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-head/-/mjml-head-4.3.0.tgz#78efcd7d46204922cc7c499775ee8f1fe4cfc082"
  integrity sha512-PBxtwak0E9EnPYZ215bN8pYSfEjJNHQExna59ezu3BDebb9luCSml9kaSJxtV7UCm9BXRaH3rK64ygFNwbCUwg==
  dependencies:
    babel-runtime "^6.26.0"
    cross-env "^5.1.4"
    lodash "^4.17.2"
    mjml-core "4.3.0"

mjml-hero@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-hero/-/mjml-hero-4.3.0.tgz#ff8bd422a9f82771fa2d643ab930eefec463121f"
  integrity sha512-scJpmN3qPWDXkJgAT/pcGmdkwXXPHHKWffjhIJS9BfuNZBAOzqNYFs9/p75YLEPtlFXyON78bdSKThh2agH73Q==
  dependencies:
    babel-runtime "^6.26.0"
    cross-env "^5.1.4"
    lodash "^4.17.2"
    mjml-core "4.3.0"

mjml-image@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-image/-/mjml-image-4.3.0.tgz#044d6c8a4cc640ac97f4b592b0788eff4c3e3b06"
  integrity sha512-ifv7hwhss0ugKYaZbuBy7Fzv5kh5N+Ae7PhNY8IPqO6k1h8p+7qyQe1ugEE3QjPtnQVRvOIufTaqq8/aISbMeQ==
  dependencies:
    babel-runtime "^6.26.0"
    cross-env "^5.1.4"
    lodash "^4.17.2"
    mjml-core "4.3.0"

mjml-migrate@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-migrate/-/mjml-migrate-4.3.0.tgz#ce73bfdb3cfea8e36ee6563849218b820f7a5381"
  integrity sha512-OgclquISA6tZsHUXQRKGoH+vMBvL6aPe9no1xC9x+otsV3c07Q8Js6RoXYqmhIfdyfbT7gGcwazyK1ziBUSsxg==
  dependencies:
    babel-runtime "^6.26.0"
    commander "^2.11.0"
    cross-env "^5.1.4"
    js-beautify "^1.6.14"
    lodash "^4.17.2"
    mjml-core "4.3.0"
    mjml-parser-xml "4.3.0"

mjml-navbar@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-navbar/-/mjml-navbar-4.3.0.tgz#b058a62a806173aee5af0ecc6c840d7eec2ed4ac"
  integrity sha512-LRMI5Yr9E6b+P9QRZHpz09wNDrOkQUVUR4KrkXH5TBUJm6VnwfPIlMzQIyJE0b8VorIJU7njRCKNtRwssDVYaA==
  dependencies:
    babel-runtime "^6.26.0"
    cross-env "^5.1.4"
    lodash "^4.17.2"
    mjml-core "4.3.0"

mjml-parser-xml@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-parser-xml/-/mjml-parser-xml-4.3.0.tgz#284603bfbbca45dc54e2d2779cceb645e9a3d4af"
  integrity sha512-/UF0NgZ3TG1pS7Rojnt8KYHR2abIrRFrbO1ilFpxYhl8HBeqze7FUfaJ6xOXrObvhTV3kJdWKHnAMyZhv02rhw==
  dependencies:
    babel-runtime "^6.26.0"
    cross-env "^5.1.4"
    htmlparser2 "^3.9.2"
    lodash "^4.17.2"

mjml-raw@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-raw/-/mjml-raw-4.3.0.tgz#7bc6f97f19d97979e5795d206cf32db04b97a9f4"
  integrity sha512-Wl12yTCJMYK3DqnW9n4STflmshUyLSqZV1ws53sORGta8HcYq/CuWWKHCKuyMDnLDQMBjZArPsgCmhq2yJRK5A==
  dependencies:
    babel-runtime "^6.26.0"
    cross-env "^5.1.4"
    lodash "^4.17.2"
    mjml-core "4.3.0"

mjml-section@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-section/-/mjml-section-4.3.0.tgz#2ec2c801f38afbb2f5d140d317026c2a2063a70b"
  integrity sha512-UJhJ1ZHKoxWfk3cbcg2djaif6V8zvlX78bTrdZvHu+nEPBRUMSRESlyDjIr+7r8eFuIDEpMSdDhHylc+f3HBHA==
  dependencies:
    babel-runtime "^6.26.0"
    cross-env "^5.1.4"
    lodash "^4.17.2"
    mjml-core "4.3.0"

mjml-social@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-social/-/mjml-social-4.3.0.tgz#14d1465df60e15ae0cea238079bb294e36e17413"
  integrity sha512-Sioy3CaaAmnhjcQMhmh0WB7XH7PQcF/CQi1SlZPUrZtwrrYGsfj+3i4LEg/EZpu5Bn9vEb39QIgb2CEGsT22Eg==
  dependencies:
    babel-runtime "^6.26.0"
    cross-env "^5.1.4"
    lodash "^4.17.2"
    mjml-core "4.3.0"

mjml-spacer@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-spacer/-/mjml-spacer-4.3.0.tgz#91492dcd11725c49d8f6fdcff057abaa5f9031ca"
  integrity sha512-Sv5r/Tovd9yQ4KSxZSLvWDDKjqLKhy+Ka9yCu4P66tw76ydWUCYAu87cSjgTGm+ALuWvshnvHH6aBzDhCJmT1Q==
  dependencies:
    babel-runtime "^6.26.0"
    cross-env "^5.1.4"
    lodash "^4.17.2"
    mjml-core "4.3.0"

mjml-table@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-table/-/mjml-table-4.3.0.tgz#86d68e893b256c529126c5cf00c205cc26729c00"
  integrity sha512-Pe5qg8qwvWCUDz+DHcRmWuryB+g8kdIVrnzhKeNU3rOxJD+Gs4YTEBL0Ml2lwwuQtTfQAMfXKrgLdE++mM5dPA==
  dependencies:
    babel-runtime "^6.26.0"
    cross-env "^5.1.4"
    lodash "^4.17.2"
    mjml-core "4.3.0"

mjml-text@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-text/-/mjml-text-4.3.0.tgz#7b43511da7ea84cc5de53a134e96f4f7d1a8507d"
  integrity sha512-FwGfaRw1fAdEA3BjW3ciTftty+WuTMZHWgj/5BbNrxpp44oxDdUHJYfVqm3GHwqcusdcHo6biJUxhSEochDixQ==
  dependencies:
    babel-runtime "^6.26.0"
    cross-env "^5.1.4"
    lodash "^4.17.2"
    mjml-core "4.3.0"

mjml-validator@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-validator/-/mjml-validator-4.3.0.tgz#f0b294fd43e58940c8ba1f9badcea75a9047af0d"
  integrity sha512-SWfsBHt4gC1zz+kjf3FWZgM9d9tdZj0VO1e0BK0Y1uqshGf192d9vMIKkqrtHI9sMkIl1Y6+/XJEjDBcbkKOvg==
  dependencies:
    babel-runtime "^6.26.0"
    cross-env "^5.1.4"
    lodash "^4.17.2"
    warning "^3.0.0"

mjml-wrapper@4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml-wrapper/-/mjml-wrapper-4.3.0.tgz#a53cab20d26840747b9d04bc158c8070b0875514"
  integrity sha512-h49VfEZHZsSdR+u8+ilGsh/kPprGTeVz4iTrv7KDkCakh1WyFKXyhfMQK0tSVd9dASfQZvC01k0mqMy/5z2/DA==
  dependencies:
    babel-runtime "^6.26.0"
    cross-env "^5.1.4"
    lodash "^4.17.2"
    mjml-core "4.3.0"
    mjml-section "4.3.0"

mjml@~4.3.0:
  version "4.3.0"
  resolved "https://registry.yarnpkg.com/mjml/-/mjml-4.3.0.tgz#16d700ddf54fe94f4431a9edbf7f63968ff87ba6"
  integrity sha512-we+WdicUut9bajevhGE0a0148hY9qauSO/9BagHcYcelSUouvJGBG6yr7XnkiZmnRG5P0QcR4wnukzXWOU6AHg==
  dependencies:
    cross-env "^5.1.4"
    mjml-accordion "4.3.0"
    mjml-body "4.3.0"
    mjml-button "4.3.0"
    mjml-carousel "4.3.0"
    mjml-cli "4.3.0"
    mjml-column "4.3.0"
    mjml-core "4.3.0"
    mjml-divider "4.3.0"
    mjml-group "4.3.0"
    mjml-head "4.3.0"
    mjml-head-attributes "4.3.0"
    mjml-head-breakpoint "4.3.0"
    mjml-head-font "4.3.0"
    mjml-head-preview "4.3.0"
    mjml-head-style "4.3.0"
    mjml-head-title "4.3.0"
    mjml-hero "4.3.0"
    mjml-image "4.3.0"
    mjml-migrate "4.3.0"
    mjml-navbar "4.3.0"
    mjml-raw "4.3.0"
    mjml-section "4.3.0"
    mjml-social "4.3.0"
    mjml-spacer "4.3.0"
    mjml-table "4.3.0"
    mjml-text "4.3.0"
    mjml-validator "4.3.0"
    mjml-wrapper "4.3.0"
```

**Comment by tkafka** [comment]
What's even weirder is that `mjml` (eg. `4.2.0`) has all it's dependencies defined as a `^4.2.0`, so that even when I specifically request to install `"mjml": "4.2.0"` (exact version), all installed components (like `mjml-body`) will be 4.3.0. This might lead to broken installs.

I propose changing dependencies of mjml to include exact component version, so that mjml 4.2.0 will always fetch 4.2.0 version of all components (and if I want eg. 4.2.1, it is up to me to have `~4.2.0` in my package.json)

**Comment by kmcb777** [comment]
This problem has been fixed in 4.3, now all the inner mjml dependencies versions will be exact 👍

**Comment by tkafka** [comment]
@iRyusa In my case I'm calling mjml like this:
```
./node_modules/.bin/mjml ./mjml-templates/$TEMPLATE.mjml -o ./html-output/$TEMPLATE.html --config.keepComments=0 --config.validationLevel=strict
```
(asking it to write to output file instead of writing to stdout) and the output file isn't created, I only get this message:
```
Error: ValidationError:
 Line 282 of /Users/kafkat/Desktop/LMC data/jobs.cz/2018 08 emaily v mjml/mjml-templates/head.mjml, included at line 5 of file /Users/kafkat/Desktop/LMC data/jobs.cz/2018 08 emaily v mjml/mjml-templates/mjml-templates/jobs-agent-aof-02-mjml.mjml (mj-raw) — mj-raw cannot be used inside mj-head, only inside: mj-column, mj-attributes, mj-body, mj-group, mj-section, mj-wrapper

Command line error:
Input file(s) failed to render
```

**Comment by iRyusa** [comment]
Yeap, for now just remove the strict validation, we'll fix it tomorrow with 4.3.1.

**Comment by tkafka** [comment]
Ok, thanks!

**Comment by donni106** [comment]
> What's even weirder is that `mjml` (eg. `4.2.0`) has all it's dependencies defined as a `^4.2.0`, so that even when I specifically request to install `"mjml": "4.2.0"` (exact version), all installed components (like `mjml-body`) will be 4.3.0. This might lead to broken installs.
> 
> I propose changing dependencies of mjml to include exact component version, so that mjml 4.2.0 will always fetch 4.2.0 version of all components (and if I want eg. 4.2.1, it is up to me to have `~4.2.0` in my package.json, so that mjml@4.2.1 will be installed, and together with it all 4.2.1 components).

Yeah! I also encountered this problem for earlier versions.

> This problem has been fixed in 4.3, now all the inner mjml dependencies versions will be exact 👍

Nice, thanks!

**Comment by bryancasler** [comment]
Adding a meta tag inside <mj-raw>, inside <mj-head>, and compiling using the VSCode extension gives me something much different. The inserted meta tag is dropped inside a <style> tag. This is the extension I'm using (https://marketplace.visualstudio.com/items?itemName=mjmlio.vscode-mjml)

<img width="1821" alt="Screen Shot 2021-01-04 at 09 19 25" src="https://user-images.githubusercontent.com/662993/103544896-c4cbed00-4e6e-11eb-8949-321124396cf0.png">
